backend/msgpack/server.go

```diff
@@ -182,12 +194,45 @@ func (s *server) write(
 		s.RUnlock()
 		return err
 	}
-	numShards := placement.NumShards()
-	shard := s.shardFn(id, numShards)
-	instances := placement.InstancesForShard(shard)
-	err = s.writerMgr.WriteTo(instances, shard, mu, pl)
+	var (
+		numShards = placement.NumShards()
+		shardID   = s.shardFn(id, numShards)
+		instances = placement.InstancesForShard(shardID)
+		nowNanos  = s.nowFn().UnixNano()
+		multiErr  = xerrors.NewMultiError()
+	)
+	for _, instance := range instances {
+		shard, ok := instance.Shards().Shard(shardID)
```

Comment:
When would this error occur? Shouldn't we be guaranteed to have the right instances by the call to `placement.InstancesForShard(shardID)`?